### DISCUSSION TITLE
Is it possible to specify different colors for same cell border?

### DISCUSSION THREAD

**Discussion by skitchpatterson3** [QuestPDF/QuestPDF#345] [Q&A] [answered]
The library allows you to specify different border widths for the border bottom, top, left, and right, as below:
```
.BorderLeft(4)
.BorderRight(5)
.BorderTop(6)
.BorderBottom(7)
```

However, I cannot seem to find a way to specify different border **colors** for the different sides, such as:
```
.BorderTop(1).BorderColor(Colors.Grey.Medium)
.BorderBottom(1).BorderColor(Colors.Black)
.BorderLeft(1).BorderColor(Colors.Blue.Medium)
.BorderRight(1).BorderColor(Colors.Yellow.Medium)
```

The result is a border with all four sides showing the last `.BorderColor` property specified (which in the example above is yellow).

Is it possible to specify different colors for some (or all) of the border sides?

**Comment by MarcinZiabek** [comment, marked as answer]
Yes, it is possible:

```csharp
.MinimalBox()
.Background(Colors.Grey.Lighten3)

.BorderRight(2)
.BorderColor(Colors.Red.Medium)

.Container() // <- important

.BorderBottom(3)
.BorderColor(Colors.Green.Medium)

.Padding(5)
.Text("Test")
.FontSize(20);
```

Produces:
![image](https://user-images.githubusercontent.com/9263853/191128735-96c1acc2-a936-4d3a-a527-c87ed34a5516.png)

**Reply by skitchpatterson3** [reply]
Fantastic! Thank you!

Playing around with it a bit more, I observed a few behaviors (noted in the in-line comments below):

```
column.Item().Width(50)
    .Background(Colors.Grey.Lighten3)
    //.Border(1).BorderColor(Colors.Black) // <- if this is uncommented, it draws a black 1-px border
                                           // around the cell which is overlaid on top of any other
                                           // borders specified below
    .Container()
    .BorderBottom(1).BorderColor(Colors.Green.Medium)

    .Container() // <- include this again to not override the green border color specified above
    .BorderTop(1).BorderColor(Colors.Blue.Medium)

    .Container() // <- include this again to not override the blue border color specified above
    .BorderLeft(1).BorderColor(Colors.Yellow.Medium)

    .Container() // <- include this again to not override the yellow border color specified above
    .BorderRight(1).BorderColor(Colors.Pink.Medium)

    .Padding(5)
    .Text("Test")
    .FontSize(20);
```
![image](https://user-images.githubusercontent.com/44105005/191136709-e7b9532d-6e69-479e-9748-8935cb3f5691.png)

Note: if the individual border side items are set to a larger width than the generic border specified at the top of the code (assuming it is uncommented), you will get a multi-colored border that has the generic border (from the top of the code) drawn over the individual side items. See below for examples:

![image](https://user-images.githubusercontent.com/44105005/191137140-aab87799-3e1b-4ce6-8073-237e9d726f07.png)
^ this has a generic border of width (1) and individual side borders of width **(2)**

![image](https://user-images.githubusercontent.com/44105005/191136989-2074c696-7fd5-4b57-aaa6-cf7a1e5692a5.png)
^ this has a generic border of width (1) and individual side borders of width **(3)**

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
